### ISSUE TITLE
Allow modules to provide content

### ISSUE THREAD

**Issue by fbrnc** [mzeis/Mzeis_Documentation#2]
It would be great if modules would be able to hook into this and provide their own pages (maybe read-only?)

So every module could come with a "user manual" that's available in this module.

**Comment by mzeis** [comment]
In fact, @tobi-pb has created an extension at the same time as me that does just that: https://github.com/code-x/Docview

We also thought about combining the two modules. I'd love to do that but at first glance I didn't see how we could merge them the best way. I'm open to suggestions how we could to that!

---

One approach to do this would be the one @tobi-pb used: have extra directories. Another one which I also could imagine would be to put the content in the config.xml or another XML file. I agree these pages should be read-only.

**Comment by mzeis** [comment]
Hi Fabrizio, I just added a prototype for this feature as [v1.1.0-beta1](https://github.com/mzeis/Mzeis_Documentation/releases/tag/v1.1.0-beta1).

There are some limitations. The biggest one is that module markdown files won't be read for most modules and developers out of the box. The reason is that often the code is symlinked into the Magento core and the documentation files are not in the `app/code/...` path, making them invisible to the extension. It may be possible to do some magic and follow the symlinks to the Composer/modman directories but I decided not to go there for several reasons. I added some information regarding this in the documentation.

Another thing I'd like to have a few people have a look at is the security of the module. Given that the module name and page name / path is provided in the URL it should be made sure that no unwanted files can be read from the file system. I added a few checks but didn't do extensive tests and I'm sure that security experts can come up with crazy stuff to test against. :-)
